### ISSUE TITLE
[Persistence] Uncaught (in promise) undefined

### ISSUE THREAD

**Issue by VWoeltjen** [nasa/openmct#1593]
Steps to reproduce (observed on the continuous deployment of WARP):

1. Create a Timeline
2. Add an Activity
3. Save As
4. Select My Items (or another directory)

The Save does complete successfully, but on console the error `Uncaught (in promise) undefined` is observed.

In practice, this appears to be innocuous, but the call stack gives me a little concern. The error occurs in [TransactionService#cancel](https://github.com/nasa/openmct/blob/ec4fe8efb39e4decb869edde9c6c61393c32ec6d/platform/commonUI/edit/src/services/TransactionService.js#L122); `transaction` is undefined in the return statement and so a rejected promise is returned. (This is already unexpected behavior.)

TransactionService#cancel is called from [EditorCapability#finish](https://github.com/nasa/openmct/blob/1bdc0497c73167d38d3a74bbe9506adacd808036/platform/commonUI/edit/src/capabilities/EditorCapability.js#L104), which does not do anything to handle promise rejection. This is potentially significant due to the clearing of the `editing` status during the promise chain.

Finally, this in turn is called from a `$destroy` listener in [EditObjectController](https://github.com/nasa/openmct/blob/1bdc0497c73167d38d3a74bbe9506adacd808036/platform/commonUI/edit/src/controllers/EditObjectController.js#L70). This looks like the root cause to me; editing concludes with a save, then an unnecessary cancel gets issued when scope gets cleared.

In practice, this appears to be innocuous, but the disruption of the promise chain is concerning. There is the potential for persistence errors and/or invalid application state, particularly as other code changes under development.

**Comment by gdunge** [comment]
I see an uncaught promise as well, whenever I save a checklist, or a todo (same code as tutorial). I haven't looked for the error when saving other objects.

--
Doug Weathers
http://www.gdunge.com
Dedicated to the creation of a spacefaring civilization

**Comment by dhrubomoy** [comment]
Looks like, after user clicks "Save and Finish Editing",  [`EditorCapability#finish()`](https://github.com/nasa/openmct/blob/master/platform/commonUI/edit/src/actions/SaveAndStopEditingAction.js#L57) gets called in `SaveAndStopEditingAction.js` file, finishing the current editing session. But then [`EditorCapability#finish()`](https://github.com/nasa/openmct/blob/1bdc0497c73167d38d3a74bbe9506adacd808036/platform/commonUI/edit/src/controllers/EditObjectController.js#L47) is called again because of [`cancelEditing(domainObject)`](https://github.com/nasa/openmct/blob/1bdc0497c73167d38d3a74bbe9506adacd808036/platform/commonUI/edit/src/controllers/EditObjectController.js#L70) getting called from the `$destroy` listener. That's why we are getting the Uncaught undefined.

If we change the [`EditorCapability#finish()`](https://github.com/nasa/openmct/blob/1bdc0497c73167d38d3a74bbe9506adacd808036/platform/commonUI/edit/src/capabilities/EditorCapability.js#L102) function to something like following, we will see `undefined` printed in the console:
```javascript
EditorCapability.prototype.finish = function () {
    var domainObject = this.domainObject;
    return this.transactionService.cancel().then(function () {
        domainObject.getCapability("status").set("editing", false);
        return domainObject;
    }).then(function(error) {
        console.log(error);
    }) ;
};
```
Thats why removing the redundant [`EditorCapability#finish()`](https://github.com/nasa/openmct/blob/1bdc0497c73167d38d3a74bbe9506adacd808036/platform/commonUI/edit/src/controllers/EditObjectController.js#L47) call from `cancelEditing()` function seems to solve the problem.

**Comment by dhrubomoy** [comment]
@VWoeltjen Should I make a pull request removing the redundant call to [`EditorCapability#finish()`](https://github.com/nasa/openmct/blob/1bdc0497c73167d38d3a74bbe9506adacd808036/platform/commonUI/edit/src/controllers/EditObjectController.js#L47) ?

**Comment by dtailor90** [comment]
Fixed in PR #1721. Closing
